### ISSUE TITLE
round-trip question

### ISSUE THREAD

**Issue by mrzhaodekui** [yuqianghan/editretro#19]
Dear author,How do you use MolecularTransformer to evaluate round-trip accuracy? Could you please give a link to the open-source code?

**Comment by yuqianghan** [comment]
The general pipeline for calculating round-trip accuracy involves the following steps:

1. Generate candidate reactants using your own model. We obtain the top 10 predictions from our model.
2. Generate candidate products based on the predicted reactants from step 1 using the pre-trained Molecular Transformer. You can refer to the "Pre-processing", "Testing", and "Evaluate Predictions" sections in https://github.com/pschwllr/MolecularTransformer. And the authors have provided the checkpoints in https://ibm.ent.box.com/v/MolecularTransformerModels.

**Comment by mrzhaodekui** [comment]
> 计算往返准确度的常规管道包括以下步骤：
> 
> 1. 使用您自己的模型生成候选反应物。我们从模型中获得前 10 个预测。
> 2. 使用预训练的 Molecular Transformer 根据步骤 1 中预测的反应物生成候选产物。您可以参考 https://github.com/pschwllr/MolecularTransformer 中的“预处理”、“测试”和“评估预测”部分。作者在 https://ibm.ent.box.com/v/MolecularTransformerModels 年提供了检查点。



> The general pipeline for calculating round-trip accuracy involves the following steps:
> 
> 1. Generate candidate reactants using your own model. We obtain the top 10 predictions from our model.
> 2. Generate candidate products based on the predicted reactants from step 1 using the pre-trained Molecular Transformer. You can refer to the "Pre-processing", "Testing", and "Evaluate Predictions" sections in https://github.com/pschwllr/MolecularTransformer. And the authors have provided the checkpoints in https://ibm.ent.box.com/v/MolecularTransformerModels.

Got it, thanks for your reply. But I noticed that Molecular Transformer has multiple checkpoints. Could you please tell me which checkpoints you chose when evaluating the round-trip?

**Comment by yuqianghan** [comment]
We used the STEREO_separated_augm_model_average_20.pt.

**Comment by mrzhaodekui** [comment]
> We used the STEREO_separated_augm_model_average_20.pt.

Got it,Thank you very much for your guidance and contributions in the field of retrosynthesis. Please accept my sincere gratitude, and I wish you even greater success in your future research endeavors.

**Comment by pansanity666** [comment]
Hi，may I ask how did you measure that   'the Ground Truth is reached' ?  By Top-1 forward exact match?

Best,

**Comment by mrzhaodekui** [comment]
> Hi，may I ask how did you measure that 'the Ground Truth is reached' ? By Top-1 forward exact match?
> 
> Best,

‘Specifically, we marked the predicted precursors as correct if the predicted precursors were the same as the ground truth or the proposed precursors yielded the target product using MT, and otherwise incorrect.’  Due to the limitations of the situation, we intend to use the method described in 'Deep Retrosynthetic Reaction Prediction using Local Reactivity and Global Attention' to calculate the round-trip accuracy, which may be different from the calculation method used by the author of this article.
